tests/layer_tests/tensorflow_tests/test_tf_Equal.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+from common.utils.tf_utils import permute_nchw_to_nhwc
+
+import logging
+
+# Testing operation Equal
+# Documentation: https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/math/equal
+
+class TestTFEqual(CommonTFLayerTest):
+    output_type = np.float32
+    x_shape = [1]
+    y_shape = [1]
+    x_value = None
+    y_value = None
+
+    # Overload inputs generation to fill dummy Equal input
+    def _prepare_input(self, inputs_dict):
+        for input in inputs_dict.keys():
+            if isinstance(self.x_shape, np.ndarray) or isinstance(self.x_shape, list):
+                if not self.x_value is None:
+                    inputs_dict[input] = np.full(inputs_dict[input], self.x_value, dtype=self.output_type).astype(self.output_type)
```

Comment:
`.astype(self.output_type)` is not needed because you specify `dtype=self.output_type`